content/sensu-go/6.8/operations/deploy-sensu/secure-postgres.md

```diff
@@ -0,0 +1,376 @@
+
+### Configure PostgreSQL to use TLS
+
+To configure your PostgreSQL instance to use TLS:
+
+1. Copy your PostgreSQL certificate files from your Sensu backend.
+From the `/etc/sensu/tls` directory, run:
+
+   {{< code shell >}}
+scp postgres.example.com* postgres.example.com:/home/user
+scp ca.pem postgres.example.com:/home/user
+{{< /code >}}
+
+2. From your PostgreSQL instance, create a new directory (`/etc/postgresql/14/main`) and paste your PostgreSQL certificate files to move them from your Sensu backend:
+
+   {{< code shell >}}
+sudo mkdir /etc/postgresql/14/main/tls
+cd /etc/postgresql/14/main/tls
+cp /home/user/postgres.example.com* .
+cp /home/user/ca.pem .
+{{< /code >}}
+
+3. Open the PostgreSQL configuration file `/etc/postgresql/14/main/postgresql.conf` in your code editor and add the following lines to enable TLS:
+
+   {{< code shell >}}
+ssl = on
+ssl_ca_file = '/etc/postgresql/14/main/tls/ca.pem'
+ssl_cert_file = '/etc/postgresql/14/main/tls/postgres.example.com.pem'
+ssl_key_file = '/etc/postgresql/14/main/tls/postgres.example.com-key.pem'
+{{< /code >}}
+
+   Save your changes and close the file.
+
+4. Open the `/etc/postgresql/14/main/pg_hba.conf` file and add the following lines to configure host-based authentication to accept certificates only when accessing the `sensu_events` database:
+
+   {{< code shell >}}
+# Prevent "postgres" superuser login via a certificate
+hostssl all             postgres        ::/0                    reject
+hostssl all             postgres        0.0.0.0/0               reject
+
+# Rules for Sensu DB connection
+hostssl sensu_events    sensu           0.0.0.0/0               cert
+{{< /code >}}
+
+5. Restart PostgreSQL:
+
+   {{< code shell >}}
+sudo systemctl restart postgresql.service
+{{< /code >}}
+
+Now that you've configured PostgreSQL to use TLS and your Sensu user is required to authenticate with a certificate, you can validate your Sensu backend configuration.
+
+### Validate Sensu backend configuration for PostgreSQL
+
+To validate that your Sensu backend can reach PostgreSQL and authenticate, run the following command:
+
+{{< code shell >}}
+curl http://localhost:8080/health
+{{< /code >}}
+
+The response should be similar to the following example.
+If the `Active` and `Healthy` attributes are not **both** `true`, stop and troubleshoot your connection to PostgreSQL *before* you continue:
+
+{{< code text >}}
+{
+  "Alarms": null,
+  "ClusterHealth": [
+    {
+      "MemberID": 13217573501179607000,
+      "MemberIDHex": "b76e4111d26d35e2",
+      "Name": "sensu.example.com",
+      "Err": "",
+      "Healthy": true
+    }
+  ],
+  "Header": {
+    "cluster_id": 11959078708079102000,
+    "member_id": 6370351775894128000,
+    "raft_term": 4242
+  },
+  "PostgresHealth": [
+    {
+      "Name": "sensu_postgres",
+      "Active": true,
+      "Healthy": true
+    }
+  ]
+}
+{{< /code >}}
+
```

Comment:
I noticed in Generate certificates, we have several warnings about deleting unneeded certificate files. See https://docs.sensu.io/sensu-go/latest/operations/deploy-sensu/generate-certificates/#generate-agent-certificate (the yellow warning) for an example. Do we need similar warnings in this doc?

Reply:
So in this case, the `ca.pem` is needed when setting up postgres tls since it's the root cert. So the warning would look something like telling folks to remove it once they've copied it over.